File: api/src/main/java/hirez/api/GitProperties.java
package hirez.api;

import java.io.IOException;
import java.util.Properties;

public class GitProperties {
    private static final Properties PROPERTIES = new Properties();

    public static final String GIT_BRANCH = "git.branch";
    public static final String GIT_COMMIT_ID = "git.commit.id";
    public static final String GIT_COMMIT_ID_ABBREV = "git.commit.id.abbrev";
    public static final String GIT_COMMIT_ID_DESCRIBE = "git.commit.id.describe";
    public static final String APPLICATION_NAME = "application.name";
    public static final String APPLICATION_VERSION = "application.version";
    public static final String APPLICATION_URL = "application.url";
    public static final String APPLICATION_DESCRIPTION = "application.description";

    public static String get(String name) {
        return PROPERTIES.getProperty(name);
    }

    static {
        try {
            PROPERTIES.load(GitProperties.class.getClassLoader().getResourceAsStream("git.properties"));
        } catch (IOException ignore) {}
    }
}

File: api/src/main/kotlin/hirez/GitProperties.kt
package hirez

import java.util.*

/**
 *
 * @author Damian Staszewski [damian@stachuofficial.tv]
 * @version %I%, %G%
 * @since 1.0
 */
object GitProperties {
	
	val APPLICATION_NAME = "application.name"
	val APPLICATION_VERSION = "application.version"
	val APPLICATION_DESCRIPTION = "application.description"
	val APPLICATION_URL = "application.url"
	val GIT_BRANCH = "git.branch"
	val GIT_COMMIT_ID = "git.commit.id"
	val GIT_COMMIT_ID_ABBREV = "git.commit.id.abbrev"
	val GIT_COMMIT_ID_DESCRIBE = "git.commit.id.describe"
	
	val PROPERTIES = Properties().apply {
		load(GitProperties::class.java.classLoader.getResourceAsStream("git.properties"))
	}
	
	fun get(key: String) =
				PROPERTIES.getProperty(key)
}